### PR TITLE
Implement Issue #250 Terraform binary tests

### PR DESCRIPTION
**Pull request by dataclouder**
This is the first part of the test suite enhancement to allow running tests using a Terraform executable tool. (Issue #250)

Add commands make test-binary and make test-binary-prepare
Add subcommands for binary testing in runtest.sh
Add script test-binary.sh with possible interactive tests
Improve production of test-artifacts scripts
Add check on test-artifacts for duplicate script names
Add tags and other metadata to test-artifacts scripts
Update TESTING.md

TODO: The second part will include the production of binary tests from custom templates.

Signed-off-by: Giuseppe Maxia <gmaxia@vmware.com>

### CODE REVIEW

File: scripts/test-binary.sh [hunk 1]
```diff
@@ -0,0 +1,209 @@
+#!/bin/bash
+
+runtime_dir=$(dirname $0)
+cd $runtime_dir
+runtime_dir=$PWD
+cd -
+pause_file=$runtime_dir/pause.txt
+dash_line="# ---------------------------------------------------------"
+
+function get_help {
```
[the remark this answers is not in the document]
Comment (dataclouder) [reply scripts/test-binary.sh]: Added

**Comment by dataclouder** [comment]
There was a missing tag in file `vcd/resource_vcd_dnat_test.go`. I missed because the file test snippets are not all at the bottom, and I stopped searching too early.
I have adjusted the tags evaluation, which was faulty, as it ran tests with `ALL` when a specific tag was requested.

File: TESTING.md [hunk 2]
```diff
@@ -152,7 +153,49 @@ For example, the unit test run as a command before the acceptance test:
...
+# OR
+./test-binary.sh pause verbose
+
+# OR
+./test-binary.sh pause verbose tags "catalog gateway"
+```
+
+The "pause" option will stop the test after every call to the terraform tool, waiting for user input.
+
+When the test runs unattended, it is possible to stop it gracefully by creating a file named `pause.txt` inside the
```
Comment (lvirbalas) [inline TESTING.md]: I suggest `pause.txt` -> `pause` filename. It's not really a text file and it's faster to type just `touch pause` when you saw an error and are in a hurry to pause.

File: TESTING.md [hunk 3]
```diff
@@ -152,7 +153,49 @@ For example, the unit test run as a command before the acceptance test:
...
+
+The test runs from GNUMakefile, using
+
+```bash
+make test-binary
+```
+
+All the tests run unattended, stopping only if there is an error.
+
+It is possible to run the tests with manual control, by preparing them and then running the test script from the `tests-artifacts` directory:
```
Comment (lvirbalas) [inline TESTING.md]: "Manual control" is a bit unclear in this context (you start to thinking about pausing the tests), so I suggest this wording instead: "It is possible to customise running of the binary tests, by ..."

File: TESTING.md (hunk 2 above)
Comment (dataclouder) [reply TESTING.md]: fixed

File: TESTING.md (hunk 3 above)
Comment (dataclouder) [reply TESTING.md]: changed

**Comment by dataclouder** [comment]
> Only one comma to remove :)

done!
